### DISCUSSION TITLE
Role base Auth for multiple tables

### DISCUSSION THREAD

**Discussion by foldess** [supabase/supabase#12772] [Questions] [answered]
Hi, i was looking around at Supabase and am currenctly a bit stuck with something.

I want to achieve some kind of role based auth.

Example: I have an app that has a TODO list, a custom virtual warehouse, and "customer profile".
Is it possible to have 3 roles and each role grants access to a table:

- todo_user
- warehouose_user
- customers_user

and than i would manually assign those roles to user.

Is this (or something similar) possible with supabase? i think it's i just did not understand how to

**Comment by GaryAustin1** [comment, marked as answer]
The two main ways I've seen being used are to have a roles table and use RLS joining on that table or using this repository: https://github.com/supabase-community/supabase-custom-claims .

Using postgres roles might be a way, but this thread discusses issues currently with that: https://github.com/supabase/supabase/discussions/11948
